src/pages/TypingTest/index.js

```diff
@@ -78,10 +78,25 @@ export default class Main extends Component {
   };
 
   backspace = () => {
-    const { index } = this.state;
+    const { index,letters } = this.state;
     if (index === 0) return;
+   
+    let newIdx = index - 1;
+    const updatedLetters = letters.map((letter, idx) => (
+      idx <= newIdx ?
+        {
+          ...letter,
+          done: true,
+        } :
+        {
+          ...letter,
+          done: false,
+        }
+    ));     
```

Comment:
We don't need to make copies of each letter, we just need to update the last one.

```javascript
    const updatedLetters = letters.map((letter, idx) => (
      idx <= newIdx ? letter : { ...letter, done: false }
    ));     
```
Though I see that you were just copying from below

Reply:
Alright, I've uploaded the changes.